### PR TITLE
[android] Enhance the "View on Map" button to extended version

### LINKED ISSUE #7650: The "show on map" button in search results is not recognisable enough
image

There've been multiple occasions when users either don't notice this important button or don't recognize its function by its looks (and never try it!). 

We had a more descriptive and bigger text "View on map" button before #2081.
AFAIR the main idea was to make it smaller to fit more search results into the screen.

However discoverability of this button is much more important than fitting one extra result.

We should make the button easily discoverable and it should be very clear what does it do.
Maybe return to the text button? Or use a bigger and better icon?

### PR DESCRIPTION
**Pull request by kavikhalique**
Fixed #7650

Changed the "View on map" icon button to Extended text icon button for ease of readability of users.
As well as corrected the night mode button text color from black to white (Reference: ios version of OM)

fix1
fix2

### CODE REVIEW

File: android/app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -105,7 +105,7 @@
   <color name="button_accent_disabled">@color/black_12</color>
   <color name="button_accent_disabled_night">#FF505050</color>
   <color name="button_accent_text">#FFFFFFFF</color>
-  <color name="button_accent_text_night">#FF000000</color>
+  <color name="button_accent_text_night">@color/white_primary</color>
```
Comment (biodranik) [inline android/app/src/main/res/values/colors.xml:108]: Why this color was changed?

File: android/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2196,4 +2196,5 @@
 	<string name="type.amenity.events_venue">Events Venue</string>
 	<string name="type.shop.auction">Auction</string>
 	<string name="type.shop.collector">Collectables</string>
+  <string name="show_on_map">View On Map</string>
```
Comment (biodranik) [inline android/app/src/main/res/values/strings.xml:2199]: This translation is already present: search_show_on_map

**Review by biodranik — COMMENTED** [review]
Thanks! Did you test it with other languages, including RTL ones?

@Jean-BaptisteC PTAL

File: android/app/src/main/res/values/colors.xml (hunk 1 above)
Comment (kavikhalique) [reply android/app/src/main/res/values/colors.xml:108]: Before the color was black in night mode for button text which was not looking good
I took reference of ios version of OM there it was white in night mode so i changed the whole button text color to white in night

I can revert it to black also

File: android/app/src/main/res/values/strings.xml (hunk 2 above)
Comment (kavikhalique) [reply android/app/src/main/res/values/strings.xml:2199]: Thankyou, I didnt saw that i will use it

**Comment by kavikhalique** [comment]
> Thanks! Did you test it with other languages, including RTL ones?
> 

Yes i have tested with arabic for RTL and one another language

It works fine.

image

image

**Comment by Misalf-git** [comment]
Black text on bright-blue actually seems better readable than white text.

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14

**Comment by kavikhalique** [comment]
> Black text on bright-blue actually seems better readable than white text.

Thanks for suggestion i reverted it back to black

**Comment by Jean-BaptisteC** [comment]
Please check all code using this FAB -> this FAB is used in [Bookmarks](https://github.com/organicmaps/organicmaps/blob/4cdb8c6ca65804e38727aac745afb3aa37205d41/android/app/src/main/java/app/organicmaps/bookmarks/BookmarksListFragment.java#L232)

**Comment by kavikhalique** [comment]
Oh i missed that, Thankyou, i will do changes and will check everywhere else also where ever FAB is used

File: android/app/src/main/res/values/colors.xml (hunk 1 above)
Comment (biodranik) [reply android/app/src/main/res/values/colors.xml]: It would be great to see screenshots from old and new versions to better understand the change.

Comment (biodranik) [reply android/app/src/main/res/values/colors.xml]: Nevermind, I see them at the bottom. Thanks!

**Review by biodranik — APPROVED** [review]
Thanks! @Jean-BaptisteC good to merge?

**Comment by Jean-BaptisteC** [comment]
Yes it's works properly in search fragment and bookmark fragment

**Review by pastk — APPROVED** [review]

**Pull request merged**